doc/languages-frameworks/coq.section.md

```diff
@@ -41,10 +45,18 @@ The recommended way of defining a derivation for a Coq library, is to use the `c
 
 It also takes other standard `mkDerivation` attributes, they are added as such, except for `meta` which extends an automatically computed `meta` (where the `platform` is the same as `coq` and the homepage is automatically computed).
 
+The builder `mkCoqDerivation` can also by applied to a function taking an attribute set and returning one. The main reason for using this feature is to get the version number as computed by `mkCoqDerivation`, as in:
+```
+cat $(nix-build -E 'with import ./. {}; coqPackages.mkCoqDerivation (computed: {
+  pname = "test"; src = mktemp; defaultVersion = "1.0";
+  VERSION = computed.version; installPhase = "echo $VERSION > $out";})')
+```
+will return `999`. This does a fixpoint computation, so do not use an attribute of `computed` in order to alter the outcome of the same attribute or you will get an infinite recursion.  For a concrete example see `coqPackage.mathcomp-analysis`.
```

Comment:
The `mathcomp-analysis` example is way more interesting *and* understandable that this one. So I would suggest quoting it directly in the doc here (but skipping the parts that are not relevant to illustrate the behavior).